### ISSUE TITLE
[App] How to get sweep information for custom chart

### ISSUE THREAD

**Issue by ethanabrooks** [wandb/client#1793]
I am trying to create a box chart that displays several metrics for each of several sweeps. I see that I can access my metrics using `summary` or `history` keys. I also see that the query accesses ID and name info, although those fields do not seem to be editable. Is there a way to access info about which sweep each run belong to, if any? Thank you.

**Comment by cvphelps** [comment]
Hi Ethan, great question. Could you please share a screenshot of what you're looking at?

**Comment by ethanabrooks** [comment]
Hi Carey, Thanks for the response. This is what I am hoping to achieve:
![image](https://user-images.githubusercontent.com/10344742/106608421-61e37980-6532-11eb-93fa-4ebc266eb681.png)

Each of the labels on the X-axis corresponds to a sweep consisting of five seeds. So for example, "CoFI-S" corresponds to sweep peyca3us. The blue bar is one metric, the orange bar another, etc.

This is as far as I've gotten with custom charts:

![image](https://user-images.githubusercontent.com/10344742/106608204-1c26b100-6532-11eb-8854-5b9da65ec3bc.png)

Once I have all the values I need (namely the sweep information), I am imagining that I can get pretty close to that first image by playing around with the Vega editor?

**Comment by KyleGoyette** [comment]
HI @ethanabrooks it sounds like it would make sense to:
1. In the runs table, group by the config column where you logged your model type, like "CoFIS" in your example, as well as the length of the instructions for that test.
2. In the query editor, query for groupKeys
3. You can then select groupKeys as the data source for the groupKeys field, and for the label field. This should Closely approximate the original chart.

If you don't have the model type in the config for each run, you'll need to add it (similarly for the instruction length). This can be done through the API:
```
import wandb
api = wandb.Api()
model_name = ...
instruction_length = ...
run = api.run("<entity>/<project>/<run_id>")
run.config["model"] = model_name
run.config["instruction_length"] = instruction_length
run.update()
```

**Comment by ethanabrooks** [comment]
Hi @KyleGoyette. Thank you for your thoughtful response. I was able to get partway through your instructions but ran into some issues:

> as well as the length of the instructions for that test.

Each run evaluates on multiple lengths, so a length corresponds to a metric, not a separate run (see image below).

> In the query editor, query for groupKeys

This is what I have right now:
![image](https://user-images.githubusercontent.com/10344742/107681237-ce662300-6c6c-11eb-938b-437feb8251f8.png)
I added the history key because it seemed necessary for the next step.

> You can then select groupKeys as the data source for the groupKeys field, and for the label field. 

This is what I have:
![image](https://user-images.githubusercontent.com/10344742/107681347-f05fa580-6c6c-11eb-883a-301749180a59.png)

The chart is close:
![image](https://user-images.githubusercontent.com/10344742/107681414-040b0c00-6c6d-11eb-886a-34c988182eb2.png)

But I would still need a way to achieve the subgroups corresponding to instruction length. Thanks for getting me this far!

**Comment by KyleGoyette** [comment]
Hi Ethan,

Please try out the spec I created below:
```{
  "$schema": "https://vega.github.io/schema/vega/v5.json",
  "description": "A basic grouped bar chart example.",
  "width": 300,
  "height": 240,
  "padding": 5,

  "data": [
    {
      "name": "wandb",
      "transform": [
        {
          "type": "formula",
          "as" : "inst_length_group",
          "expr": "if(datum['${field:position}']<6, '0-6', '6+')"
        }
      ]
    }
  ],
  "legends": [
    {"stroke": "color", "symbolType": "stroke", "title": "Instruction Length"}
  ],

  "scales": [
    {
      "name": "yscale",
      "type": "band",
      "domain": {"data": "wandb", "field": "${field:category}"},
      "range": "height",
      "padding": 0.2
    },
    {
      "name": "xscale",
      "type": "linear",
      "domain": {"data": "wandb", "field": "${field:value}"},
      "range": "width",
      "round": true,
      "zero": true,
      "nice": true
    },
    {
      "name": "color",
      "type": "ordinal",
      "domain": {"data": "wandb", "field": "inst_length_group"},
      "range": {"scheme": "category20"}
    }
  ],

  "axes": [
    {"orient": "left", "scale": "yscale", "tickSize": 0, "labelPadding": 4, "zindex": 1, "title": "${field:category}"},
    {"orient": "bottom", "scale": "xscale", "title": "${field:value}"}
  ],
  "marks": [
    {
      "type": "group",

      "from": {
        "facet": {
          "data": "wandb",
          "name": "facet",
          "groupby": "${field:category}"
        }
      },

      "encode": {
        "enter": {
          "y": {"scale": "yscale", "field": "${field:category}"}
        }
      },

      "signals": [
        {"name": "height", "update": "bandwidth('yscale')"}
      ],

      "scales": [
        {
          "name": "pos",
          "type": "band",
          "range": "height",
          "domain": {"data": "facet", "field": "inst_length_group"}
        }
      ],

      "marks": [
        {
          "name": "bars",
          "from": {"data": "facet"},
          "type": "rect",
          "encode": {
            "enter": {
              "y": {"scale": "pos", "field": "inst_length_group"},
              "height": {"scale": "pos", "band": 1},
              "x": {"scale": "xscale", "field": "${field:value}"},
              "x2": {"scale": "xscale", "value": 0},
              "fill": {"scale": "color", "field": "inst_length_group"}
            }
          }
        }
      ]
    }
  ]
}```


I made this with the following query and field mapping:
![image](https://user-images.githubusercontent.com/13547142/107695832-faca7100-6c65-11eb-84b4-0d7fde532ed1.png)

I couldn't find a runset with the archiectures, but I think you can replaced seed with your architecture config and it will work.

**Comment by ethanabrooks** [comment]
Kyle, thanks so much for your help on this. The chart doesn't compile due to some slight differences in my data, but I think it's enough to get me going. I'll post back on here as soon as I have the solution.

**Comment by KyleGoyette** [comment]
Great, thanks Ethan. I believe as long as you can query for your architecture (say in a config for a run), you should be able use that as your category field mapping.

**Comment by ethanabrooks** [comment]
Kyle, I was able to get your implementation to run. I just had to follow your instructions a little more closely 🤦‍♂️ The graph does not quite look right because `instruction length` is aggregated across multiple runs, not logged separately for each run and as a result it does not properly differentiate among them. Maybe that is something I will change in my code. Anyway, this is enough to get me going. Thanks!
